### PR TITLE
DEVPROD-7734 Add notice level for project validation

### PR DESCRIPTION
**Pull request by ZackarySantana**
DEVPROD-7734

### Description
Adds notice level for project validation. This also changes [DEVPROD-6656](https://jira.mongodb.org/browse/DEVPROD-6656) to be of level notice rather than warning

### Testing
Staging test and add some unit tests modifications.

Staging example:

```
zackary.santana % evg-staging validate evergreen.yml -w

NOTICE: buildvariant 'zzztest-tag1-small' references distro 'zzztest-tag1-small' with the following admin-defined warning: This is a warning. If it is actually blocking, you can remove it.
evergreen.yml is valid with warnings
```

### Documentation
I'm not sure if we have docs currently explaining error vs warning. I _think_ it's self explanitory even with notices, but not sure what reviewer(s) might think! I couldn't find anywhere this would fit in the docs, I'm cool with making new docs for it though if yall think!

### CODE REVIEW

File: validator/project_validator.go [hunk 1]
```diff
@@ -116,6 +119,16 @@ func (v ValidationErrors) HasError() bool {
 	return false
 }
 
+// Has returns if any of the errors are at the given level.
+func (v ValidationErrors) Has(level ValidationErrorLevel) bool {
+	for _, err := range v {
```
Comment (Kimchelly) [inline validator/project_validator.go:124]: nit: to reduce copy-pasting, might be simpler to just return `len(v.AtLevel(level)) > 0`.

File: validator/project_validator.go [hunk 2]
```diff
@@ -116,6 +119,16 @@ func (v ValidationErrors) HasError() bool {
 	return false
 }
```
Comment (Kimchelly) [inline validator/project_validator.go:120]: Can we remove this and replace with `Has(Error)`? It's duplicate functionality now.

File: operations/validate.go [hunk 3]
```diff
@@ -143,10 +143,12 @@ func validateFile(path string, ac *legacyClient, quiet, includeLong, errorOnWarn
 	}
 
 	grip.Info(projErrors)
-	if projErrors.HasError() || (errorOnWarnings && len(projErrors) > 0) {
+	if projErrors.Has(validator.Error) || (errorOnWarnings && projErrors.Has(validator.Warning)) {
 		return errors.Errorf("%s is an invalid configuration", path)
-	} else if len(projErrors) > 0 {
+	} else if projErrors.Has(validator.Warning) {
 		grip.Infof("%s is valid with warnings", path)
```
Comment (Kimchelly) [inline operations/validate.go:149]: Might want to say "with warnings/notices" in case there are both.

**Review by Kimchelly — CHANGES_REQUESTED** [review]
Don't forget to bump the CLI version!

File: validator/project_validator.go (hunk 2 above)
Comment (ZackarySantana) [reply validator/project_validator.go:120]: Yes! I actually removed all instances already and meant to delete this- ty for the catch

File: operations/validate.go [hunk 4]
```diff
@@ -143,10 +143,12 @@ func validateFile(path string, ac *legacyClient, quiet, includeLong, errorOnWarn
 	}
 
 	grip.Info(projErrors)
-	if projErrors.HasError() || (errorOnWarnings && len(projErrors) > 0) {
+	if projErrors.Has(validator.Error) || (errorOnWarnings && projErrors.Has(validator.Warning)) {
 		return errors.Errorf("%s is an invalid configuration", path)
-	} else if len(projErrors) > 0 {
-		grip.Infof("%s is valid with warnings", path)
+	} else if projErrors.Has(validator.Warning) {
+		grip.Infof("%s is valid with warnings/notices", path)
```
Comment (ablack12) [inline operations/validate.go:149]: Personally I think we can say this for both the warning and the notice case

**Review by ablack12 — APPROVED** [review]
🚀

**Review by Kimchelly — APPROVED** [review]

File: validator/project_validator.go (hunk 1 above)
Comment (ZackarySantana) [reply validator/project_validator.go]: I redid it to the one you suggested! I don't think it'll be a bottleneck either but I have heard of the horror stories that some projects have hundreds of warnings 😨

Comment (Kimchelly) [reply validator/project_validator.go]: It's true that some projects have a lot of warnings and this is technically less efficient, but in practice, iterating in-memory through hundreds of items is so fast that it really won't make a difference. It's really only doing something expensive in a loop (e.g. DB calls in a loop) that needs optimization.

**Pull request merged**